catalog/resource_sql_table.go

```diff
@@ -22,12 +23,23 @@ import (
 var MaxSqlExecWaitTimeout = 50
 
 type SqlColumnInfo struct {
-	Name     string `json:"name"`
-	Type     string `json:"type_text,omitempty" tf:"alias:type,computed"`
-	Comment  string `json:"comment,omitempty"`
-	Nullable bool   `json:"nullable,omitempty" tf:"default:true"`
+	Name     string         `json:"name"`
+	Type     string         `json:"type_text,omitempty" tf:"alias:type,computed"`
+	Identity IdentityColumn `json:"identity,omitempty"`
+	Comment  string         `json:"comment,omitempty"`
+	Nullable bool           `json:"nullable,omitempty" tf:"default:true"`
+	TypeJson string         `json:"type_json,omitempty" tf:"computed"`
 }
 
+type TypeJson struct {
+	Metadata map[string]any `json:"metadata,omitempty"`
+}
+
+type IdentityColumn string
+
+const IdentityColumnAlways IdentityColumn = "always"
+const IdentityColumnDefault IdentityColumn = "default"
```

Comment:
Maybe for clarity we can have a constant to indicate no identity column?
```suggestion
const IdentityColumnDefault IdentityColumn = "default"
const IdentityColumnNone IdentityColumn = ""
```